core/tests/mod.rs

```diff
@@ -0,0 +1,646 @@
+
+        let sumeragis_arc = Arc::clone(&sumeragis);
+
+        let local_post_proc = Arc::clone(&post_proc);
+        let sumeragi = SumeragiHandle::start_closure(SumeragiStartClosureArgs {
+            configuration: &configuration.sumeragi,
+            events_sender,
+            wsv,
+            queue: Arc::clone(&queue),
+            kura,
+            genesis_network: GenesisNetwork::test(
+                configuration.clone(),
+                index == 0 && commit_genesis,
+            ),
+            block_hashes: &[],
+            post_procedure: Arc::new(move |packet, peer_id| {
+                local_post_proc(packet, peer_id, &sumeragis_arc);
+            }),
+            update_topology_procedure: Arc::new(move |_network_topology| {}),
+            get_connected_peers_procedure: Arc::new(HashSet::new),
+        });
+
+        sumeragis.lock().unwrap().insert(peer_id.clone(), sumeragi);
+        queues.insert(peer_id.clone(), queue);
+    }
+    (keypairs, peer_ids, configs, queues, sumeragis)
+}
+
+fn init_telemetry() {
+    iroha_logger::init(&blank_config().logger).expect("Failed to initialize telemetry");
+}
+
+fn blank_config() -> Configuration {
+    iroha::samples::get_config_proxy(HashSet::new(), Some(KeyPair::generate().unwrap()))
+        .build()
+        .unwrap()
+}
+
+/// Get a standardised key-pair from the hard-coded literals.
+///
+/// # Panics
+/// Programmer error. Hardcoded keys must be in proper format.
+pub fn get_key_pair() -> KeyPair {
+    KeyPair::new(
+        PublicKey::from_str(
+            "ed01207233BFC89DCBD68C19FDE6CE6158225298EC1131B6A130D1AEB454C1AB5183C0",
+        )
+        .expect("Public key not in mulithash format"),
+        PrivateKey::from_hex(
+            Algorithm::Ed25519,
+            "9AC47ABF59B356E0BD7DCBBBB4DEC080E302156A48CA907E47CB6AEA1D32719E7233BFC89DCBD68C19FDE6CE6158225298EC1131B6A130D1AEB454C1AB5183C0".as_ref(),
+        ).expect("Private key not hex encoded")
+    ).expect("Key pair mismatch")
+}
+
+static GENESIS_KEY: Lazy<KeyPair> = Lazy::new(|| KeyPair::generate().unwrap());
+
+fn genesis_account(public_key: iroha_crypto::PublicKey) -> Account {
+    Account::new(AccountId::genesis(), [public_key]).build(&AccountId::genesis())
+}
+
+fn genesis_world(peers: Vec<PeerId>) -> World {
+    let account_public_key = GENESIS_KEY.public_key().clone();
+    let mut domain = Domain::new(DomainId::genesis()).build(&AccountId::genesis());
+
+    domain.accounts.insert(
+        <Account as Identifiable>::Id::genesis(),
+        genesis_account(account_public_key),
+    );
+
+    World::with(vec![domain], peers)
+}
+
+fn sleep_ms(t: u64) {
+    std::thread::sleep(std::time::Duration::from_millis(t));
+}
+
+/// Trait used to differentiate a test instance of `genesis`.
+pub trait TestGenesis: Sized {
+    /// Construct Iroha genesis network and optionally submit genesis
+    /// from the given peer.
+    fn test(cfg: Configuration, submit_genesis: bool) -> Option<Self>;
+}
```

Comment:
This reminds me of code that we removed last year. I still think that we ought to avoid mocking and instead try to integration test what we can. 

@Erigara is right in that the scope of the current tests is bigger than unit tests, but smaller than integration. This lies in the uncanny valley of placebo testing: it's not utilising the full code paths and requires a lot of effort to maintain, but it pushes coverage up significantly. 

If we can't test core without resorting to such tricks, then maybe we ought to refine our work on proper integration and functional tests.